Ziko's Joint, a chatbot

vrill opened the pull request:
Here's my contribution to this fun and challenging project. Not too imaginative, but with full focus on the code and getting the functions down, I hope upcoming projects can lend some more time and resources toward the feel of them. Enjoy!

vrill committed:
Initial commit.

Did and un-did changes to script.js, is now in original state.

s

vrill committed:
i:test
test
test

vrill committed:
Base logic in place

I've styled the CSS slightly, and introduced the base logic for the bot
to get through the first set of questions and feedback. About halfway
there, with optimizations being part of the latter stretch.

vrill committed:
Logic in place

Need to finalize last reload loop, and final styling remains to be done.

vrill committed:
v0.3

Logic fully in place, with every bot/user interaction working as intended. Stylizing
and optimization to be done.§# Please enter the commit message for your changes. Lines starting

vrill committed:
v0.4

Missed the final response from the user, now all logis is in place... :)

vrill committed:
v0.5

Code (JS) clean-up.

vrill committed:
Read me

Detailed project description.

vrill committed:
Delete .gitignore

vrill committed:
Removed folder

.vscode folder deleted from repository.

vrill committed:
Merge branch 'master' of https://github.com/vrill/project-p4-chatbot

vrill committed:
New pics

My choice of pic of bot and user.

vrill committed:
v1.0

Final push, at least I feel done. May do minor changes, but I am happy
with the result. Enjoy :)

HIPPIEKICK commented on code/script.js:
```diff
@@ -1,5 +1,47 @@
+const inputWrapper = document.getElementById('input-wrapper')
+
+const exitHTML = `
+<button id='yes' type='submit'>✔ Yes</button>
+<button id='no' type='submit'>❌ No</button>
+`
+const paymentHTML = `
+<button id='card' type='submit'>💳 By card</button>
+<button id='cash' type='submit'>💸 With cash</button>
+<button id='dishes' type='submit'>🍽 I'll do the dishes</button>
+`
+const selectPizza = `
+<select id='select'>
+<option value='' selected disabled>Which one will it be?</option>
+<option value='pizza blanco'>Pizza Blanco</option>
+<option value='svennebananpizza'>Svennebanan</option>
+<option value='honey and chevre pizza'>Honey Chevre</option>
+</select>
+`
+const selectPasta = `
+<select id='select'>
+<option value='' selected disabled>Which one would you like?</option>
+<option value='salsiccia bolognese pasta'>Salsiccia bolognese</option>
+<option value='shrimp and saffron pasta'>Shrimp & saffron</option>
+<option value='creamy fungi pasta'>Creamy fungi</option>
+</select>
+`
+const selectSalad = `
+<select id='select'>
+<option value='' selected disabled>Which one will clench your thirst?</option>
+<option value='cobb salad'>Cobb salad</option>
+<option value='caesar salad'>Caesar salad</option>
+<option value='feta and beetroot salad'>Feta & beetroot</option>
+</select>
+`
+const buttonSelectMain = `
+<button id='pizza' type='submit'>🍕 Pizza</button>
+<button id='pasta' type='submit'>🧆 Pasta</button>
+<button id='salad' type='submit'>🥗 Salad</button>
+`
```
Super nice that you created variables for the HTML content, looks 👌
To follow conventions and improve readability, you could indent your content like this:
```
const exitHTML = `
  <button id='yes' type='submit'>✔ Yes</button>
  <button id='no' type='submit'>❌ No</button>
`
```

or even like this (when you have more than one attribute):
```
const exitHTML = `
  <button
    id='yes'
    type='submit'>
    ✔ Yes
  </button>
  <button
    id='no'
    type='submit'>
    ❌ No
  </button>
`
```

If you want to learn more about conventions - take a look at [our code guidelines](https://www.notion.so/technigo/Guidelines-for-how-to-write-good-code-69ffb59eac4e4358a7c5752efb873c4e).

HIPPIEKICK commented on code/script.js:
```diff
@@ -17,9 +60,10 @@ const showMessage = (message, sender) => {
+    `
       <section class="bot-msg">
-        <img src="assets/bot.png" alt="Bot" />
+        <img src="assets/bot.png" alt="Ziko" />
```
Nice that you changed the images 🤤

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+// Interaction 1, initial greeting.
+// Ask user for name, and store name. Wait 1 second before greeting user.
+const greeting = () => {
+  showMessage(`Hello and welcome to Ziko's joint. What's your name?`, 'bot')
```
Minor thing, but since this string doesn't include any JavaScript variables, you could use single quotes

vrill replied:
@HIPPIEKICK Will this work when I have a single quote in the sentence itself, such as "Ziko`'`s or "Let`'`s"? VS Code underlines it, expecting commas.
I've changed to `'` where applicable (v1.1) as per your suggestion, otherwise I've left the ` for these situations.

HIPPIEKICK replied:
You're absolutely right in leaving it this way 👍

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
-setTimeout(greeting, 1000)
+// Interaction 2, exchange names.
+// Use the stored name, and as user reply with name. Pass stored name on.
+const handleNameInput = (event) => {
```
Minor thing, this parameter doesn't need parentheses around it (it's only when you have zero arguments or multiple arguments that you need them):
`const handleNameInput = event => {`

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+
+// Interaction 3, looking at the menu.
+// As bot, greet the user with their name. Then present the choices the user may select. Store choice and pass it on.
+const showMainOptions = (userName) => {
+  showMessage(
+    `Nice to meet you, ${userName}. Which of our lovely main options would you like to choose?`,
+    'bot')
+  inputWrapper.innerHTML = buttonSelectMain
+  document
+  .getElementById('pizza')
+  .addEventListener('click', () => {
+    showMessage(`Let's see which pizzas you have!`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => mainChoice('pizza'), 1000)
+  })
+  document
+  .getElementById('pasta')
+  .addEventListener('click', () => {
+    showMessage(`I'd love me some pasta!`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => mainChoice('pasta'), 1000)
+  })
+  document
+  .getElementById('salad')
+  .addEventListener('click', () => {
+    showMessage(`Today I'll go with a salad.`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => mainChoice('salad'), 1000)
+  })
+}
```
Clean code with all the new lines 👌

HIPPIEKICK replied:
And great name of the function!

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+// Interaction 4, choose what to eat.
+// As bot, confirm choice. Depending on choice, show specific menu the user may select as a drop-down. Pass the value of this choice on.
+const mainChoice = (specificChoice) => {
+  if (specificChoice === 'pizza') {
+  showMessage(`Excellent, pizza's always a winner! Make your choice below.`, 'bot')
+  inputWrapper.innerHTML = selectPizza
+  } else if (specificChoice === 'pasta') {
```
Your menu variables keep the `mainChoice` function really clean ⭐ You can indent the code in the conditionals as well, like this:
```
if (specificChoice === 'pizza') {
  showMessage(`Excellent, pizza's always a winner! Make your choice below.`, 'bot')
  inputWrapper.innerHTML = selectPizza
}
```

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+  showMessage(`Fresh, clean and tasty! Make your choice below.`, 'bot')
+  inputWrapper.innerHTML = selectSalad
+  }
+  const dish = document.getElementById('select')
+  addEventListener('change', () => {
+    showMessage(`Okay. I'll go with a ${dish.value}, thanks.`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => payment(dish.value), 1000)
+  })
```
Nice with a drop-down ⭐

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+  yes.addEventListener('click', () => {
+    showMessage(`Yup, GET IN MY BELLY!`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => done(yes), 1000)
```
You don't need to pass an argument to the `done` function :)

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
-// Set up your eventlisteners here
+// Interaction 4, choose what to eat.
+// As bot, confirm choice. Depending on choice, show specific menu the user may select as a drop-down. Pass the value of this choice on.
+const mainChoice = (specificChoice) => {
```
Maybe this function can be called `showSpecificMenu` or similar to better explain what the function does?

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+
+// Interaction 5, food decision has been made.
+// As bot, reinforce choice. Ask user for payment, and present options the user may choose from. Pass the choice of payment on.
+const payment = (dish) => {
```
And this could be called `showPaymentOptions` or similar

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+
+// Interaction 6, final confirmation.
+// As bot, tell user they've chosen how to pay and that it's time to confirm wether to go through with order (if so: define, store & pass on decision), or to start over (go back to the beginning).
+const exitStrategy = (trade) => {
```
Suggestion on name: `askForConfirmation`

HIPPIEKICK commented on code/script.js:
```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
-  // Just to check it out, change 'bot' to 'user' here 👆
+// Interaction 7, win-win.
+// As bot, hand over food item to user and wish them a happy meal (what's the English phrase?)
+const done = () => {
```
Naming suggestion: `wishHappyMeal` or `showLastMessage` or `sayBuonAppetito` (I don't know the English phrase either 🤓 )

HIPPIEKICK reviewed (commented):
Great job with the chatbot Christoffer! I like that you did your own thing with the styling. You write clean code and your comments are really helpful; they make your code become even easier to read. This will be of value as we progress to more complex projects ⭐ Keep up the good work!

vrill committed:
v1.1

change: removed an unnecessary variable, and improve indent
change: used single quotation marks where applicable
change: rename a few functions for clarity
